File: API/WorkoutTracker.Domain/BaseClasses/Entity.cs
using System;
using System.Collections.Generic;
using System.Text;
using WorkoutTracker.Domain.Interfaces;

namespace WorkoutTracker.Domain.BaseClasses
{
    public abstract class Entity : IEntity
    {
        public int Id { get; set; }
        public int CreatedByUserId { get; set; }
        public DateTime CreatedDateTime { get; set; }
        public int? ModifiedByUserId { get; set; }
        public DateTime? ModifiedDateTime { get; set; }
    }
}

File: workout-tracker/src/app/shared/models/entity.ts
import { IMightHaveAuditDates } from "../interfaces/i-might-have-audit-dates";
export abstract class Entity implements IMightHaveAuditDates {
  id: number = 0;
  createdByUserId: number | undefined;
  createdDateTime: Date | undefined;
  modifiedByUserId: number | null = null;
  modifiedDateTime: Date | null = null;
}
